### PR TITLE
8313765: Invalid CEN header (invalid zip64 extra data field size)

### PR DESCRIPTION
**Pull request by benty-amzn**
Backport is not clean, required changes:
- Some context lines
- Changes to `CorruptedZipFiles.java` had to be altered because [8304014](https://bugs.openjdk.org/browse/JDK-8304014) has not been backported to jdk20
- Path in `ReadNonStandardExtraHeadersTest.java` had to be converted to URI, I believe because of FileSystems API changes in JDK13

All tests in `test/jdk/java/util/zip` pass

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8313765](https://bugs.openjdk.org/browse/JDK-8313765): Invalid CEN header (invalid zip64 extra data field size) (**Bug** - P2)

### Reviewers
 * [Andrew John Hughes](https://openjdk.org/census#andrew) (@gnu-andrew - **Reviewer**)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev.git pull/2084/head:pull/2084` \
`$ git checkout pull/2084`

Update a local copy of the PR: \
`$ git checkout pull/2084` \
`$ git pull https://git.openjdk.org/jdk11u-dev.git pull/2084/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2084`

View PR using the GUI difftool: \
`$ git pr show -t 2084`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/2084.diff">https://git.openjdk.org/jdk11u-dev/pull/2084.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk11u-dev/pull/2084#issuecomment-1681515007)

### CODE REVIEW

**Comment by benty-amzn** [comment]
Failing linux GHA looks like infra issues - https://github.com/benty-amzn/jdk11u-dev/actions/runs/5885841578/job/15962854580#step:6:76

**Comment by simonis** [comment]
> > Failing linux GHA looks like infra issues - https://github.com/benty-amzn/jdk11u-dev/actions/runs/5885841578/job/15962854580#step:6:76
> 
> Yes, this looks like a variant of the same problem I fixed for 8u in [JDK-8284772](https://bugs.openjdk.org/browse/JDK-8284772), namely that the very specific version of gcc used is fragile and easily broken by Ubuntu updates. I'd suggest applying something similar to 11u. We do need this to run through GHA before being committed.

@gnu-andrew are you working on fixing GHA for jdk11u and is there a JBS issue for it already or should we pick it up?

**Comment by gnu-andrew** [comment]
> > > Failing linux GHA looks like infra issues - https://github.com/benty-amzn/jdk11u-dev/actions/runs/5885841578/job/15962854580#step:6:76
> > 
> > 
> > Yes, this looks like a variant of the same problem I fixed for 8u in [JDK-8284772](https://bugs.openjdk.org/browse/JDK-8284772), namely that the very specific version of gcc used is fragile and easily broken by Ubuntu updates. I'd suggest applying something similar to 11u. We do need this to run through GHA before being committed.
> 
> @gnu-andrew are you working on fixing GHA for jdk11u and is there a JBS issue for it already or should we pick it up?

I'm about to open a PR, which this one can then depend on and retest. It may be contentious though, as I intend to suggest dropping the minor version requirement altogether rather than continually bumping it every time Ubuntu update the gcc package. We did this in 8u some time ago and it's not been an issue since.

Note that this PR will need the same changes we added to the 17u one, as 11u is also still on JTReg 6.

**Comment by gnu-andrew** [comment]
https://github.com/openjdk/jdk11u-dev/pull/2087

This is already progressing further in GHA than the run here managed. If you make this dependent on that PR (change the target from openjdk:master to openjdk:pr/2087) it should be possible to re-run the tests

**Comment by benty-amzn** [comment]
Updated. Would you like me to also include the commit on from [jdk17u-dev #1670](https://github.com/openjdk/jdk17u-dev/pull/1670) to allow running on jtreg6?

**Comment by gnu-andrew** [comment]
> Updated. Would you like me to also include the commit from [jdk17u-dev #1670](https://github.com/openjdk/jdk17u-dev/pull/1670) to allow running on jtreg6?

Yes please. Looks like we're now seeing the same failure as we did in 17u.

File: test/jdk/java/util/zip/ZipFile/ReadNonStandardExtraHeadersTest.java [hunk 1]
```diff
@@ -0,0 +1,939 @@
...
+        zipFilesToTest().forEach(path -> {
+            try {
+                test.zipFilesToTest(path);
+                test.readZipFSTest(path);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+}
```
Comment (mrserb) [inline test/jdk/java/util/zip/ZipFile/ReadNonStandardExtraHeadersTest.java:939]: The new line?

**Review by gnu-andrew — APPROVED** [review]
Looks good to me. Nice that we can get rid of the need to wrap `GetBooleanAction` with `AccessController.doPrivileged` thanks to the switch to `GetPropertyAction` and `privilegedGetProperty`.

**Comment by gnu-andrew** [comment]
I imagine the GHA change will only be picked up when the fix goes into `jdk11u-dev/master` , unless the branch itself is actually rebased on top of pr/2084

**Comment by benty-amzn** [comment]
Would you like me to do that? Not sure if that will complicate the merge

**Comment by simonis** [comment]
Wouldn't it be enough to push https://github.com/openjdk/jdk11u-dev/pull/2087? This PR depends on #2087 so once that is pushed it should be easy to merge it into this PR. Or am I missing something?

**Comment by gnu-andrew** [comment]
> Wouldn't it be enough to push #2087? This PR depends on #2087 so once that is pushed it should be easy to merge it into this PR. Or am I missing something?

Yes, that's what I was thinking. I didn't realise @GoeLin was back and had approved it until just now. I've integrated it, so we should be able to just rebase this onto `jdk11u-dev/master`.

**Review by mrserb — APPROVED** [review]

**Comment by gnu-andrew** [comment]
The bug has `jdk11u-fix-yes` now, so if you integrate, one of us can sponsor and get this in.

**Pull request closed without merging**